### PR TITLE
SONAR-7345 Rename "debt" to "effort" and "effort to fix" to "gap"

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: sonar-batch/src/main/java/org/sonar/batch/issue/TrackedIssueAdapter.java [hunk 1]
```diff
@@ -144,11 +150,17 @@ public String actionPlanKey() {
   }
 
   @Override
+  @Deprecated
   public Duration debt() {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/issue/TrackedIssueAdapter.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)
MAJOR Add the missing @deprecated Javadoc tag. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)

File: sonar-batch/src/main/java/org/sonar/batch/issue/TrackedIssueAdapter.java [hunk 2]
```diff
@@ -69,7 +69,13 @@ public Integer line() {
   }
 
   @Override
+  @Deprecated
   public Double effortToFix() {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/issue/TrackedIssueAdapter.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)
MAJOR Add the missing @deprecated Javadoc tag. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)

File: sonar-batch/src/main/java/org/sonar/batch/issue/DeprecatedIssueWrapper.java [hunk 3]
```diff
@@ -82,7 +82,13 @@ public Integer line() {
   }
 
   @Override
+  @Deprecated
   public Double effortToFix() {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/issue/DeprecatedIssueWrapper.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)
MAJOR Add the missing @deprecated Javadoc tag. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)

File: server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueDoc.java [hunk 4]
```diff
@@ -217,9 +224,16 @@ public boolean isNew() {
 
   @Override
   @CheckForNull
+  @Deprecated
   public Duration debt() {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueDoc.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)
MAJOR Add the missing @deprecated Javadoc tag. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)

File: server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueDoc.java [hunk 5]
```diff
@@ -117,10 +117,17 @@ public Integer line() {
     return getNullableField(IssueIndexDefinition.FIELD_ISSUE_LINE);
   }
 
+  @Deprecated
   @Override
   @CheckForNull
   public Double effortToFix() {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueDoc.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)
MAJOR Add the missing @deprecated Javadoc tag. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)

File: server/sonar-server/src/main/java/org/sonar/server/issue/ws/SearchAction.java [hunk 6]
```diff
@@ -128,41 +131,42 @@ public void define(WebService.NewController controller) {
...
       .setPossibleValues(IssueIndex.SUPPORTED_FACETS);
-    action.createParam(IssueFilterParameters.FACET_MODE)
-      .setDefaultValue(IssueFilterParameters.FACET_MODE_COUNT)
-      .setDescription("Choose the returned value for facet items, either count of issues or sum of debt.")
-      .setPossibleValues(IssueFilterParameters.FACET_MODE_COUNT, IssueFilterParameters.FACET_MODE_DEBT);
+    action.createParam(FACET_MODE)
+      .setDefaultValue(FACET_MODE_COUNT)
+      .setDescription("Choose the returned value for facet items, either count of issues or sum of debt.<br/>" +
+        "Since 5.5, 'debt' mode is deprecated and replaced by 'effort'")
+      .setPossibleValues(FACET_MODE_COUNT, FACET_MODE_EFFORT, DEPRECATED_FACET_MODE_DEBT);
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/issue/ws/SearchAction.java]: MINOR Remove this use of "DEPRECATED_FACET_MODE_DEBT"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: sonar-plugin-api/src/main/java/org/sonar/api/issue/Issue.java [hunk 7]
```diff
@@ -186,12 +195,18 @@
   boolean isNew();
 
   /**
-   * @since 5.0
+   * @deprecated since 5.5, replaced by {@link #effort()}
    */
-  @CheckForNull
+  @Deprecated
   Duration debt();
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/issue/Issue.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/issue/Issue.java [hunk 8]
```diff
@@ -106,6 +106,13 @@
   Integer line();
 
   /**
+   * @deprecated since 5.5, replace by {@link #gap()}
+   */
+  @Deprecated
+  @CheckForNull
+  Double effortToFix();
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/issue/Issue.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: server/sonar-server/src/main/java/org/sonar/server/issue/IssueUpdater.java [hunk 9]
```diff
@@ -269,14 +273,14 @@ public boolean setEffortToFix(DefaultIssue issue, @Nullable Double d, IssueChang
 
   public boolean setPastEffortToFix(DefaultIssue issue, @Nullable Double previousEffort, IssueChangeContext context) {
     Double currentEffort = issue.effortToFix();
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/issue/IssueUpdater.java]: MINOR Remove this use of "effortToFix"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: server/sonar-server/src/main/java/org/sonar/server/issue/IssueUpdater.java [hunk 10]
```diff
@@ -287,7 +291,7 @@ public boolean setTechnicalDebt(DefaultIssue issue, @Nullable Duration value, Is
 
   public boolean setPastTechnicalDebt(DefaultIssue issue, @Nullable Duration previousTechnicalDebt, IssueChangeContext context) {
     Duration currentTechnicalDebt = issue.debt();
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/issue/IssueUpdater.java]: MINOR Remove this use of "debt"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: server/sonar-server/src/main/java/org/sonar/server/issue/IssueUpdater.java [hunk 11]
```diff
@@ -257,7 +261,7 @@ public void setCloseDate(DefaultIssue issue, @Nullable Date d, IssueChangeContex
 
   public boolean setEffortToFix(DefaultIssue issue, @Nullable Double d, IssueChangeContext context) {
     if (!Objects.equal(d, issue.effortToFix())) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/issue/IssueUpdater.java]: MINOR Remove this use of "effortToFix"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: server/sonar-server/src/main/java/org/sonar/server/issue/IssueUpdater.java [hunk 12]
```diff
@@ -269,14 +273,14 @@ public boolean setEffortToFix(DefaultIssue issue, @Nullable Double d, IssueChang
 
   public boolean setPastEffortToFix(DefaultIssue issue, @Nullable Double previousEffort, IssueChangeContext context) {
     Double currentEffort = issue.effortToFix();
-    issue.setEffortToFix(previousEffort);
+    issue.setGap(previousEffort);
     return setEffortToFix(issue, currentEffort, context);
   }
 
   public boolean setTechnicalDebt(DefaultIssue issue, @Nullable Duration value, IssueChangeContext context) {
     Duration oldValue = issue.debt();
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/issue/IssueUpdater.java]: MINOR Remove this use of "debt"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: sonar-plugin-api/src/main/java/org/sonar/api/ce/measure/test/TestIssue.java [hunk 13]
```diff
@@ -80,12 +80,13 @@ public String severity() {
 
   @Override
   @CheckForNull
+  @Deprecated
   public Duration debt() {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/ce/measure/test/TestIssue.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)
MAJOR Add the missing @deprecated Javadoc tag. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)

File: sonar-plugin-api/src/main/java/org/sonar/api/ce/measure/test/TestIssue.java [hunk 14]
```diff
@@ -80,12 +80,13 @@ public String severity() {
...
-    return debt;
+    return effort();
   }
 
   public Duration effort() {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/ce/measure/test/TestIssue.java]: MAJOR Add the "@Override" annotation above this method signature [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1161)

File: sonar-plugin-api/src/main/java/org/sonar/api/ce/measure/test/TestIssue.java [hunk 15]
```diff
@@ -127,8 +128,16 @@ public Builder setRuleKey(RuleKey ruleKey) {
       return this;
     }
 
+    /**
+     * @deprecated since 5.5, use {@link #setEffort(Duration)} instead
+     */
+    @Deprecated
     public Builder setDebt(@Nullable Duration debt) {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/ce/measure/test/TestIssue.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java [hunk 16]
```diff
@@ -575,7 +593,7 @@ private void addAssignedToMeFacetIfNeeded(SearchRequestBuilder builder, SearchOp
 
   private AggregationBuilder createResolutionFacet(IssueQuery query, Map<String, FilterBuilder> filters, QueryBuilder esQuery) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java]: MAJOR Make "createResolutionFacet" a "static" method. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2325)

File: server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java [hunk 17]
```diff
@@ -381,77 +398,78 @@ private void configureStickyFacets(IssueQuery query, SearchOptions options, Map<
...
-      aggregation.subAggregation(DEBT_AGGREGATION);
+  private static AggregationBuilder addEffortAggregationIfNeeded(IssueQuery query, AggregationBuilder aggregation) {
+    if (hasQueryEffortFacet(query)) {
+      aggregation.subAggregation(EFFORT_AGGREGATION);
     }
     return aggregation;
   }
 
+  private static boolean hasQueryEffortFacet(IssueQuery query) {
+    return FACET_MODE_EFFORT.equals(query.facetMode()) || DEPRECATED_FACET_MODE_DEBT.equals(query.facetMode());
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java]: MINOR Remove this use of "DEPRECATED_FACET_MODE_DEBT"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java [hunk 18]
```diff
@@ -599,7 +617,7 @@ private AggregationBuilder createResolutionFacet(IssueQuery query, Map<String, F
 
   private AggregationBuilder createActionPlansFacet(IssueQuery query, Map<String, FilterBuilder> filters, QueryBuilder esQuery) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java]: MAJOR Make "createActionPlansFacet" a "static" method. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2325)

File: sonar-db/src/main/java/org/sonar/db/issue/IssueDto.java [hunk 19]
```diff
@@ -109,8 +109,8 @@ public static IssueDto toDtoForComputationInsert(DefaultIssue issue, int ruleId,
       .setLine(issue.line())
       .setLocations((DbIssues.Locations) issue.getLocations())
       .setMessage(issue.message())
-      .setEffortToFix(issue.effortToFix())
-      .setDebt(issue.debtInMinutes())
+      .setGap(issue.effortToFix())
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/issue/IssueDto.java]: MINOR Remove this use of "effortToFix"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: sonar-db/src/main/java/org/sonar/db/issue/IssueDto.java [hunk 20]
```diff
@@ -158,8 +158,8 @@ public static IssueDto toDtoForUpdate(DefaultIssue issue, long now) {
...
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/issue/IssueDto.java]: MINOR Remove this use of "effortToFix"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: sonar-ws/src/main/java/org/sonarqube/ws/client/issue/IssueFilterParameters.java [hunk 21]
```diff
@@ -71,7 +71,12 @@
   public static final String FACET_MODE = "facetMode";
   public static final String FACET_MODE_COUNT = "count";
 
-  public static final String FACET_MODE_DEBT = "debt";
+  /**
+   * @deprecated since 5.5, replaced by {@link #FACET_MODE_EFFORT}
+   */
+  @Deprecated
+  public static final String DEPRECATED_FACET_MODE_DEBT = "debt";
```
Comment (SonarTech) [inline sonar-ws/src/main/java/org/sonarqube/ws/client/issue/IssueFilterParameters.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: sonar-core/src/main/java/org/sonar/core/issue/DefaultIssue.java [hunk 22]
```diff
@@ -267,34 +267,49 @@ public DefaultIssue setLine(@Nullable Integer l) {
     return this;
   }
 
+  @Deprecated
   @Override
   @CheckForNull
   public Double effortToFix() {
```
Comment (SonarTech) [inline sonar-core/src/main/java/org/sonar/core/issue/DefaultIssue.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)
MAJOR Add the missing @deprecated Javadoc tag. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)

File: sonar-core/src/main/java/org/sonar/core/issue/DefaultIssue.java [hunk 23]
```diff
@@ -267,34 +267,49 @@ public DefaultIssue setLine(@Nullable Integer l) {
...
+  public DefaultIssue setGap(@Nullable Double d) {
+    Preconditions.checkArgument(d == null || d >= 0, format("Gap must be greater than or equal 0 (got %s)", d));
+    this.gap = d;
     return this;
   }
 
+  @Deprecated
+  @Override
+  @CheckForNull
+  public Duration debt() {
```
Comment (SonarTech) [inline sonar-core/src/main/java/org/sonar/core/issue/DefaultIssue.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)
MAJOR Add the missing @deprecated Javadoc tag. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)

File: sonar-plugin-api/src/main/java/org/sonar/api/ce/measure/Issue.java [hunk 24]
```diff
@@ -51,9 +51,18 @@
    */
   String severity();
 
-  @CheckForNull
+  /**
+   * @deprecated since 5.5, replaced by {@link #effort()}
+   */
+  @Deprecated
   Duration debt();
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/ce/measure/Issue.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: sonar-batch/src/main/java/org/sonar/batch/issue/DeprecatedIssueAdapterForFilter.java [hunk 25]
```diff
@@ -82,7 +81,13 @@ public Integer line() {
   }
 
   @Override
+  @Deprecated
   public Double effortToFix() {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/issue/DeprecatedIssueAdapterForFilter.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: sonar-batch/src/main/java/org/sonar/batch/issue/DeprecatedIssueAdapterForFilter.java [hunk 26]
```diff
@@ -151,8 +156,14 @@ public boolean isNew() {
     throw unsupported();
   }
 
+  @Deprecated
   @Override
   public Duration debt() {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/issue/DeprecatedIssueAdapterForFilter.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 36 issues:
* MAJOR 11 major
* MINOR 10 minor
* INFO 15 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Remove this use of "effortToFix"; it is deprecated.](https://github.com/SonarSource/sonarqube/blob/4b1bc8269a0053624a9389b95ef38e86ebb7e833/server/sonar-server/src/main/java/org/sonar/server/computation/issue/IssueLifecycle.java#L95) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Remove this use of "debt"; it is deprecated.](https://github.com/SonarSource/sonarqube/blob/4b1bc8269a0053624a9389b95ef38e86ebb7e833/server/sonar-server/src/main/java/org/sonar/server/computation/issue/IssueLifecycle.java#L96) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by teryk** [comment]
In the UI, debt is visible in the issues page

debt_visible_in_ui

**Comment by julienlancelot** [comment]
@teryk Feedbacks applied.

File: sonar-plugin-api/src/main/java/org/sonar/api/issue/Issue.java [hunk 27]
```diff
@@ -106,6 +106,13 @@
   Integer line();
 
   /**
+   * @deprecated since 5.5, replaced by {@link #gap()}
+   */
+  @Deprecated
+  @CheckForNull
+  Double effortToFix();
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/issue/Issue.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/ce/measure/test/TestIssue.java [hunk 28]
```diff
@@ -78,16 +78,27 @@ public String severity() {
     return severity;
   }
 
+  /**
+   * @deprecated since 5.5, replaced by {@link #effort}
+   */
   @Override
   @CheckForNull
+  @Deprecated
   public Duration debt() {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/ce/measure/test/TestIssue.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: sonar-db/src/main/java/org/sonar/db/issue/IssueDto.java (hunk 20 above)
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/issue/IssueDto.java]: MINOR Remove this use of "effortToFix"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 34 issues:
* MAJOR 9 major
* MINOR 10 minor
* INFO 15 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Remove this use of "effortToFix"; it is deprecated.](https://github.com/SonarSource/sonarqube/blob/5dd15afbb51d95fc1b1f38985138d3e334fa1a20/server/sonar-server/src/main/java/org/sonar/server/computation/issue/IssueLifecycle.java#L95) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Remove this use of "debt"; it is deprecated.](https://github.com/SonarSource/sonarqube/blob/5dd15afbb51d95fc1b1f38985138d3e334fa1a20/server/sonar-server/src/main/java/org/sonar/server/computation/issue/IssueLifecycle.java#L96) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: sonar-db/src/main/java/org/sonar/db/issue/IssueDto.java (hunk 20 above)
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/issue/IssueDto.java]: MINOR Remove this use of "effortToFix"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 34 issues:
* MAJOR 9 major
* MINOR 10 minor
* INFO 15 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Remove this use of "effortToFix"; it is deprecated.](https://github.com/SonarSource/sonarqube/blob/a31bd47b02fb3c51b5fcc1c71c4453a808ac988a/server/sonar-server/src/main/java/org/sonar/server/computation/issue/IssueLifecycle.java#L95) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Remove this use of "debt"; it is deprecated.](https://github.com/SonarSource/sonarqube/blob/a31bd47b02fb3c51b5fcc1c71c4453a808ac988a/server/sonar-server/src/main/java/org/sonar/server/computation/issue/IssueLifecycle.java#L96) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by teryk** [comment]
lgtm-jesus

File: sonar-batch/src/main/java/org/sonar/batch/issue/TrackedIssueAdapter.java [hunk 29]
```diff
@@ -68,8 +68,17 @@ public Integer line() {
     return issue.startLine();
   }
 
+  /**
+   * @deprecated since 5.5, replaced by {@link #gap()}
+   */
   @Override
+  @Deprecated
   public Double effortToFix() {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/issue/TrackedIssueAdapter.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: sonar-batch/src/main/java/org/sonar/batch/issue/TrackedIssueAdapter.java [hunk 30]
```diff
@@ -143,12 +152,21 @@ public String actionPlanKey() {
     return new ArrayList<>();
   }
 
+  /**
+   * @deprecated since 5.5, replaced by {@link #effort()}
+   */
   @Override
+  @Deprecated
   public Duration debt() {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/issue/TrackedIssueAdapter.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: sonar-batch/src/main/java/org/sonar/batch/issue/DeprecatedIssueWrapper.java [hunk 31]
```diff
@@ -81,8 +81,17 @@ public Integer line() {
     return textRange != null ? textRange.start().line() : null;
   }
 
+  /**
+   * @deprecated since 5.5, replaced by {@link #gap()}
+   */
   @Override
+  @Deprecated
   public Double effortToFix() {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/issue/DeprecatedIssueWrapper.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueDoc.java [hunk 32]
```diff
@@ -215,11 +225,21 @@ public boolean isNew() {
     throw new IllegalStateException("isNew is only available for batch");
   }
 
+  /**
+   * @deprecated since 5.5, replaced by {@link #effort()}
+   */
   @Override
   @CheckForNull
+  @Deprecated
   public Duration debt() {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueDoc.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueDoc.java [hunk 33]
```diff
@@ -117,10 +117,20 @@ public Integer line() {
     return getNullableField(IssueIndexDefinition.FIELD_ISSUE_LINE);
   }
 
+  /**
+   * @deprecated since 5.5, replaced by {@link #gap()}
+   */
+  @Deprecated
   @Override
   @CheckForNull
   public Double effortToFix() {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueDoc.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: sonar-core/src/main/java/org/sonar/core/issue/DefaultIssue.java [hunk 34]
```diff
@@ -267,34 +267,55 @@ public DefaultIssue setLine(@Nullable Integer l) {
     return this;
   }
 
+  /**
+   * @deprecated since5.5, replaced by {@link #gap()}
+   */
+  @Deprecated
   @Override
   @CheckForNull
   public Double effortToFix() {
```
Comment (SonarTech) [inline sonar-core/src/main/java/org/sonar/core/issue/DefaultIssue.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: sonar-core/src/main/java/org/sonar/core/issue/DefaultIssue.java [hunk 35]
```diff
@@ -267,34 +267,55 @@ public DefaultIssue setLine(@Nullable Integer l) {
...
   }
 
   /**
-   * Elapsed time to fix the issue
+   * @deprecated since5.5, replaced by {@link #effort()}
    */
+  @Deprecated
   @Override
   @CheckForNull
   public Duration debt() {
```
Comment (SonarTech) [inline sonar-core/src/main/java/org/sonar/core/issue/DefaultIssue.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: server/sonar-server/src/main/java/org/sonar/server/computation/issue/IssueLifecycle.java [hunk 36]
```diff
@@ -92,8 +92,8 @@ public void mergeExistingOpenIssue(DefaultIssue raw, DefaultIssue base) {
     updater.setPastLine(raw, base.getLine());
     updater.setPastLocations(raw, base.getLocations());
     updater.setPastMessage(raw, base.getMessage(), changeContext);
-    updater.setPastEffortToFix(raw, base.effortToFix(), changeContext);
-    updater.setPastTechnicalDebt(raw, base.debt(), changeContext);
+    updater.setPastGap(raw, base.effortToFix(), changeContext);
+    updater.setPastEffort(raw, base.debt(), changeContext);
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/issue/IssueLifecycle.java]: MINOR Remove this use of "debt"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: server/sonar-server/src/main/java/org/sonar/server/computation/issue/IssueLifecycle.java [hunk 37]
```diff
@@ -92,8 +92,8 @@ public void mergeExistingOpenIssue(DefaultIssue raw, DefaultIssue base) {
     updater.setPastLine(raw, base.getLine());
     updater.setPastLocations(raw, base.getLocations());
     updater.setPastMessage(raw, base.getMessage(), changeContext);
-    updater.setPastEffortToFix(raw, base.effortToFix(), changeContext);
-    updater.setPastTechnicalDebt(raw, base.debt(), changeContext);
+    updater.setPastGap(raw, base.effortToFix(), changeContext);
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/issue/IssueLifecycle.java]: MINOR Remove this use of "effortToFix"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 19 issues:
* MINOR 4 minor
* INFO 15 info

Watch the comments in this conversation to review them.

**Pull request merged**